### PR TITLE
Refactor module layout

### PR DESCRIPTION
**Pull request by blink1073**
Fixes https://github.com/jupyterlab/jupyterlab/issues/3263#issuecomment-345697674.

### CODE REVIEW

**Comment by blink1073** [comment]
ping @tacaswell, mind turning on Travis for this repo?   https://travis-ci.org/profile/matplotlib

**Comment by blink1073** [comment]
Build is passing on my fork: https://travis-ci.org/blink1073/jupyter-matplotlib/builds/304846065.

File: js/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "watch": "webpack --watch"
   },
   "jupyterlab": {
-    "extension": "src/labplugin"
+    "extension": "src/lab_plugin"
```
Comment (vidartf) [inline js/package.json]: The file was renamed to `lab_extension` not `lab_plugin`. Am I missing something?

Comment (blink1073) [reply js/package.json]: No, I was ;).  I fixed it in a follow up.

**Pull request merged**